### PR TITLE
Creating Tests base and GetContainer test for RS Vault OneSDK

### PR DESCRIPTION
**Pull request by dragonfly91**
Creating Tests base and GetContainer test for RS Vault OneSDK

### CODE REVIEW

File: src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/RecoveryServicesTestBase.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+using Microsoft.Azure.Management.RecoveryServices.Backup;
+using Microsoft.Azure.Management.RecoveryServices.Backup.Models;
+using Microsoft.Azure.Test;
+using Microsoft.Azure.Test.HttpRecorder;
+
+namespace RecoveryServices.Tests
+{
+    public class RecoveryServicesTestsBase : TestBase
+    {
+        public static T GetServiceClient<T>(string resourceNamespace) where T : class
```
Comment (sudhakar4171) [inline src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/RecoveryServicesTestBase.cs:49]: do you want to duplicate all these methods across backupvault and RSvault? How about creating some common which both projects reference

File: src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/ContainerTests.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace RecoveryServices.Tests
+{
+    public class ContainerTests : RecoveryServicesTestsBase
+    {
+        [Fact]
+        public void ListContainersTest()
```
Comment (sudhakar4171) [inline src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/ContainerTests.cs]: Just think when MAB and DPM tests comes in future, do they need to duplicate all that code you write for IaaS? do we need to put some common logic in separate files

File: src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/ContainerTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            try
+            {
+                using (UndoContext context = UndoContext.Current)
+                {
+                    context.Start();
+                    var client = GetServiceClient<RecoveryServicesBackupManagementClient>("Microsoft.RecoveryServicesINTD");
+
+                    ProtectionContainerListQueryParams queryParams = new ProtectionContainerListQueryParams();
+                    queryParams.ProviderType = "AzureIaasVM";
+                    var response = client.Container.List("ASRRG1", "backupvault1", queryParams, GetCustomRequestHeaders());
```
Comment (sudhakar4171) [inline src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/ContainerTests.cs]: do you want to create common constants file and define any hard-coded values

**Comment by dragonfly91** [comment]
Updated the PR by incorporating the suggestions given by Sudreddy

File: src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/AzureIaasVMTests/ContainerTests.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+{
+    public class ContainerTests : RecoveryServicesTestsBase
+    {
+        [Fact]
+        public void ListContainersTest()
+        {
+            ExecuteTest(
+                client =>
+                {
+                    string rsVaultRgName = GetSetting(TestConstants.RsVaultRgName);
```
Comment (sudhakar4171) [inline src/ResourceManagement/AzureBackup/RecoveryServices.Tests/ScenarioTests/AzureIaasVMTests/ContainerTests.cs]: If our intention is to use same vault for all tests, then it is better to move this common logic as well into containerhelper.cs.

**Pull request merged**